appinventor/buildserver/src/com/google/appinventor/buildserver/tasks/android/CreateManifest.java

```diff
@@ -108,6 +109,17 @@ public TaskResult execute(AndroidCompilerContext context) {
         }
       }
 
+      final Map<String, Set<String>> featuresNeeded = context.getComponentInfo().getFeaturesNeeded();
+      if (!featuresNeeded.isEmpty()) {
+        final Set<String> uniqueFeatures = new HashSet<>();
+        for (Map.Entry<String, Set<String>> componentSubElSetPair : featuresNeeded.entrySet()) {
+          uniqueFeatures.addAll(componentSubElSetPair.getValue());
```

Comment:
How are you planning to resolve if two extensions have the annotation but one says required=false and another says required=true? I don't feel like the semantics are really captured in this PR. We should also understand how it interacts with the `uses-feature` declarations that are inserted above.